docs/data/charts/funnel/funnel.md

```diff
@@ -1,15 +1,111 @@
 
-:::warning
-The Funnel Chart component isn't available yet, but you can upvote [**this GitHub issue**](https://github.com/mui/mui-x/issues/7929) to see it arrive sooner.
+## Basics
 
-Don't hesitate to leave a comment there to influence what gets built.
-Especially if you already have a use case for this component, or if you're facing a pain point with your current solution.
-:::
+The funnel accepts a series which must have a data property containing an array of objects. Those objects should contain a property `value`. They can also have other optional properties, like `label` and `id`.
+
+{{"demo": "BasicFunnel.js"}}
+
+### Display legends
+
+The funnel chart displays a legend by default. The only requirement is to provide a `label` value in the data objects.
+
+To disable the legend, set the `hideLegend` property to `true`.
+
+{{"demo": "FunnelLegend.js"}}
+
+## Labels
+
+The funnel chart displays labels by default. It shows the `value` of the data item in each section.
+To format the labels, a `valueFormatter` function can be provided.
+
+{{"demo": "FunnelLabels.js"}}
+
+### Positioning labels
+
+The labels can be positioned relative to the section they represent. The `sectionLabel` property accepts an object with the following properties:
+
+- `position.vertical`: The vertical position of the label. It can be `top`, `center` or `bottom`.
+- `position.horizontal`: The horizontal position of the label. It can be `start`, `middle` or `end`.
+- `margin`: The margin between the label and the section.
+- `dominantBaseline`: The vertical alignment of the text. It can be `auto`, `baseline`, `text-before-edge`, `text-after-edge`, `central`, `hanging` or `middle`.
+- `textAnchor`: The horizontal alignment of the text. It can be `start`, `middle` or `end`.
+
+The `sectionLabel` property can be set to `false` to hide the labels.
+It also accepts a function that receives the data object and should return the label configuration.
+
+{{"demo": "FunnelLabelPositioningNoSnap.js"}}
+
+## Curve interpolation
+
+The interpolation between data points can be customized by the `curve` property.
+This property expects one of the following string values, corresponding to the interpolation method: `'linear'`, `'bumpX'`, `'bumpY'` and `'step'`.
+
+This series property adds the option to control the interpolation of a series.
+Different series could even have different interpolations.
+
+{{"demo": "FunnelCurvesNoSnap.js"}}
+
+## Colors
+
+The funnel colors can be customized in two ways.
+
+1. You can provide a [color palette](/x/react-charts/styling/#color-palette). Each section of the funnel will be colored according to this palette.
+2. You can provide a `color` property in `data` objects which overrides the palette.
+
+```jsx
+<FunnelChart
+  colors={['red', 'blue', 'green']} // Use palette
+  series={[
+    {
+      data: [
+        { value: 10, color: 'orange' }, // Use color property
+        // ...
+      ],
+    },
+  ]}
+/>
+```
```

Comment:
Could the code snipet and the demo be merged in a single one? Or it's the one that can not be small enough to have a preview?

```suggestion
The following code snippet defines colors by a red, blue, green cycle.
But the second appears orange because it's object override the `color` propoerty.

```jsx
<FunnelChart
  colors={['red', 'blue', 'green']}
  series={[{
    data: [
        { value: 20 }, // Get color from the palette (red)
        { value: 10, color: 'orange' }, // Override color (bleu replaced by orange)
        { value: 5 }, // Get color from the palette (green)
        // ...
      ],
    }]}
/>
```
```